### PR TITLE
[protobuf] upgrade to latest release (v3.19.4)

### PR DESCRIPTION
**Pull request by sphawk**
**Describe the pull request**

- #### What does your PR fix?  
  upgrade to latest release (v3.19.4)

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  unchanged, no

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  yes

 most of the code from #21485 

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

**Comment by JonLiu1993** [comment]
@sphawk, thanks for your PR, CI tests fail on many triples, but not all of them are related to your PR, I found that x64-osx triples test failures are related to your PR
Here is the error log, please see:
[install-x64-osx-dbg-out.log](https://github.com/microsoft/vcpkg/files/8221343/install-x64-osx-dbg-out.log)

**Comment by sphawk** [comment]
@JonLiu1993 sorry for late response. I'm ready for review
take most of code #21485 + solve x64-windows-static-md problem.

File: ports/protobuf/fix-static-build.patch [hunk 1]
```diff
@@ -1,13 +1,30 @@
...
+@@ -182,7 +182,11 @@ else (protobuf_BUILD_SHARED_LIBS)
+   # making programmatic control difficult.  Prefer the functionality in newer
+   # CMake versions when available.
+   if(CMAKE_VERSION VERSION_GREATER 3.15 OR CMAKE_VERSION VERSION_EQUAL 3.15)
+-    set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreaded$<$<CONFIG:Debug>:Debug>)
++    if (MSVC AND protobuf_MSVC_STATIC_RUNTIME)
++      set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreaded$<$<CONFIG:Debug>:Debug>)
++    elseif (MSVC AND NOT protobuf_MSVC_STATIC_RUNTIME)
++      set(CMAKE_MSVC_RUNTIME_LIBRARY MultiThreaded$<$<CONFIG:Debug>:Debug>DLL)
++    endif ()
```
Comment (ras0219-msft) [inline ports/protobuf/fix-static-build.patch:14]: These should already be set via the triplet & toolchain, so we should be able to simply not set CMAKE_MSVC_RUNTIME_LIBRARY at all.

**Review by ras0219-msft — COMMENTED** [review]
Thanks for the PR!

File: ports/protobuf/fix-static-build.patch (hunk 1 above)
Comment (sphawk) [reply ports/protobuf/fix-static-build.patch:14]: ok, I'll try remove all

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Feature 'zlib' tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static

**Pull request merged**